File: js/easy/q6.js
/*
 * You are a space cadet that has found a robot; whose sorting function is mismatching,
 * The robot cannot see for sure; and is mixing all its position coordinates that render his vision!
 * 
 * You are assigned to sort those position coordinates in order to help the robot!
 * 
 * Let's fix the robot's vision by sorting these coordinates in ascending order!
 */

function bubbleSort(coordinates) {
    let n = coordinates.length;
    for (let i = 0; i < n; i++) {  
        for (let j = 0; j < n - i; j++) {  
            if (coordinates[j] > coordinates[j + 1]) {
                
                [coordinates[j], coordinates[j + 1]] = [coordinates[j + 1], coordinates[j]];
            }
        }
    }
}

function printCoordinates(coordinates) {
    console.log("Robot's sorted position coordinates: ");
    coordinates.forEach(coordinate => console.log(coordinate));  
}

const robotCoordinates = [7, 3, 9, 2, 5, 4]; 

console.log("Mission: Sorting the robot's coordinates to help it see clearly!");

bubbIeSort(robotCoordinates);

printCoordinates(robotCoordinates);  

File: c-cpp/easy/q6.cpp
#include <iostream>
#include <vector>
using namespace std::count, std::endl;

void bubbleSort(vector<int>& coordinates) {
    int n = coordinates.size();
    
    // Space Cadet starts the mission, but the robot's vision is clouded.
    
    for (int i = 0; i < n; i++) {  // 
        for (int j = 0; j < n - i; j++) {  
            if (coordinates[j] > coordinates[j + 1]) {
                swap(coordinates[j], coordinates[j + 1]);
            }
        }
    }
}

void printCoordinates(const vector<int>& coordinates) {
    cout << "Robot's sorted position coordinates: ";
    for (int coordinate : coordinates) {
        cout << coordinate << "\n";
    }
    cout << endl;
}

int main() {
    vector<int> robotCoordinates = {7, 3, 9, 2, 5, 4};  
    
    cout << "Mission: Sorting the robot's coordinates to help it see clearly!" << endl;

    bubbleSort(robotCoordinates);

    printCoordinates(robotCoordinates);  
    
    return 0;
}
